Update to 1.20.1

Vexicle opened the pull request:
## Summary

<!-- What is this pull request for? Does it fix any issues? -->
This pull request attempts to implement AltoClef to run with 1.20.1 minecraft.
## Information

<!-- Put an x inside [ ] to check it, like so: [x] -->

- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, examples, ...).

## Checklist

<!-- Put an x inside [ ] to check it, like so: [x] -->

- [x] I have searched the open pull requests for duplicates.
- [ ] If code changes were made then they have been tested.

Vexicle committed:
Update gradle.properties

Need to add new blocks to support the new update, and also need to do this for baritone fork

JustaSqu1d commented on gradle.properties:
```diff
@@ -8,13 +8,13 @@ org.gradle.caching=true
+yarn_mappings=1.20.1+build.9
+loader_version=0.14.21
 # Mod Properties
 mod_version=0.4.0
```
```suggestion
mod_version=0.5.0
```

Vexicle replied:
ah sorry, still learning java

JustaSqu1d reviewed (changes requested):
Please fix the compilation errors when trying to compile Alto Clef.

Note: upgrading to the next Minecraft version isn't as simple as changing a few numbers.

Vexicle committed:
Update gradle.properties

Co-authored-by: JustaSqu1d <89910983+JustaSqu1d@users.noreply.github.com>

Vexicle committed:
Update altoclef.mixins.json

Vexicle committed:
Attempted to add smithing templates to TaskCatalogue.java

Vexicle committed:
Update gradle distro to 8.2.1

Vexicle committed:
Add smithing template recipe

still need to find a way to get the bot to look through chests first to get the smithing template, might have to look at the gamer command because the bot goes through chests in the beginning.

Vexicle committed:
preparation of implementing a new task

I will add more soon
